include/boost/winasio/http/http_url_handler.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef BOOST_WINASIO_HTTP_URL_HANDLER_HPP
+#define BOOST_WINASIO_HTTP_URL_HANDLER_HPP
+
+#if defined(_MSC_VER) && (_MSC_VER >= 1200)
+#pragma once
+#endif // defined(_MSC_VER) && (_MSC_VER >= 1200)
+
+#include <boost/winasio/http/http_major_version.hpp>
+#include <boost/log/trivial.hpp>
+#include <boost/asio/error.hpp>
+
+
+#include <string>
+
+namespace boost {
+namespace winasio {
+namespace http {
+
+template <HTTP_MAJOR_VERSION> class url_handler {};
+
+template <> class url_handler<HTTP_MAJOR_VERSION::http_ver_1> {
+public:
+  url_handler(HANDLE queue_handler) : queue_handler_(queue_handler) {}
+  void add_url(std::wstring const &url, system::error_code &ec) {
+    DWORD retCode = HttpAddUrl(queue_handler_, // Req Queue
+                               url.c_str(),    // Fully qualified URL
+                               NULL            // Reserved
+    );
+    // std::wcout << L"added url " << url << std::endl;
+    ec = system::error_code(retCode, asio::error::get_system_category());
+  }
+
+  void remove_url(std::wstring const &url, system::error_code &ec) {
+    DWORD retCode = HttpRemoveUrl(queue_handler_, // Req Queue
+                                  url.c_str()     // Fully qualified URL
+    );
+    // std::wcout << L"removed url " << url << std::endl;
+    ec = system::error_code(retCode, asio::error::get_system_category());
+  }
+  void set_queue_handler(HANDLE queue_handler, system::error_code &ec) {
+    queue_handler_ = queue_handler;
+  }
+
+private:
+  HANDLE queue_handler_ = nullptr;
+};
+
+template <> class url_handler<HTTP_MAJOR_VERSION::http_ver_2> {
+public:
+  url_handler(HANDLE queue_handler) : queue_handler_(queue_handler) {
+    DWORD retCode = HttpCreateServerSession(HTTPAPI_VERSION_2, &session_id_, 0);
```

Comment:
If user wants to use session for something else or wants to create 2 sessions. This might be restrictive, because you always create a session in url_handler.
But this is good enough for our current simple use case.

Reply:
Not familiar with the use cases, was thinking to hide the complexity of session/url_group/url/queue from the caller.  If multiple sessions and multiple groups are allowed, then more complicated handling logic for creation/removal (maybe switch) needs to add. Agree to defer it until use cases are defined.